src/firefly/js/charts/ui/XYPlotOptions.jsx

```diff
@@ -107,7 +107,8 @@ export function resultsSuccess(callback, flds, tblId) {
         shading: flds.shading || undefined,
         x : { columnOrExpr : xName, error: xErr, label : xLabel, unit : xUnit, options : xOptions},
         y : { columnOrExpr : yName, error: yErr, label : yLabel, unit : yUnit, options : yOptions},
-        tblId
+        tblId,
+        zoom,
```

Comment:
You are right.  When the columns change, the zoom should not be preserved. Thanks for pointing it out.  I did not realize it.